<?xml version="1.0" ?>
<sdf version="1.6">
  <world name="detachable_joint">
    <plugin filename="libignition-gazebo-physics-system.so"
      name="ignition::gazebo::systems::Physics"/>

    <model name="ground_plane">
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
            </plane>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="M1">
      <pose>0 0 1 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>1</mass>
          <inertia>
            <ixx>0.667</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.667</iyy>
            <iyz>0</iyz>
            <izz>0.667</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <box>
              <size>2.0 2.0 2.0</size>
            </box>
          </geometry>
        </collision>
      </link>

      <plugin filename="libignition-gazebo-detachable-joint-system.so" name="ignition::gazebo::systems::DetachableJoint">
        <parent_model_link>body</parent_model_link>
        <child_model>M2</child_model>
        <child_model_link>body</child_model_link>
      </plugin>
    </model>

    <model name="M2">
      <pose>0 0 5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>1</mass>
          <inertia>
            <ixx>0.667</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.667</iyy>
            <iyz>0</iyz>
            <izz>0.667</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <box>
              <size>2.0 2.0 2.0</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>
  </world>
</sdf>
